src/firstscreen.cpp

```diff
@@ -44,19 +45,28 @@ FirstScreen::FirstScreen( PackageBackend *backend, QString *tmpFileName, const Q
 
     //Add Repository
     int i = 0;
+    m_numOfRepositories = 0;
+    m_numOfPackages = 0;
     QVBoxLayout *repoDetails;
     m_untrustedSources = 0;
-    foreach( OCI::Repository *iter, m_repos) {
-        m_backend->addRepository( QUrl( iter->url() ) );
+    foreach( OCI::Repository *repo, m_repos) {
+        // Only proceed if it is the recommended repository
+        if(repo->recommended() == "false")
+            continue;
+        ZypperUtils::initRepository(repo->url().toStdString());
+        ++m_numOfRepositories;
+        m_backend->addRepository( QUrl( repo->url() ) );
         RepositoryWidget *repositoryDetails = new RepositoryWidget( m_backend, i, m_repos.at( i ) );
         mainLayout->addWidget( repositoryDetails );
-        if( !m_backend->exists( iter->url() ) )
+        if( !m_backend->exists( repo->url() ) )
             m_untrustedSources++;
         static int j = 0;
```

Comment:
I just noticed this. Can you make this variable static as well, but part of the class instead of local to this method? something like s_packageID ? and then use new PackageDetails( package, s_packageID++, m_packages.count() );